### ISSUE TITLE
Federate the api/v1/status/tsdb endpoint from Prometheus

### ISSUE THREAD

**Issue by fpetkovski** [thanos-io/thanos#5395]
### Is your proposal related to a problem?

It would be great if Thanos Query could federate the `api/v1/status/tsdb` endpoint from Prometheus instances. This will allow users to analyze cardinality on the fly.

### Describe the solution you'd like

Thanos Query could fanout to Prometheus sidecars and return a federated result from all endpoints. 

### Describe alternatives you've considered

We currently run https://github.com/thought-machine/prometheus-cardinality-exporter, but this requires ingesting metrics which further increases cardinality :)

**Comment by bwplotka** [comment]
Yes, I would see that as a potential gRPC API for our endpoints that Querier could merge. 

> We currently run https://github.com/thought-machine/prometheus-cardinality-exporter, but this requires ingesting metrics which further increases cardinality :)

Not if you use separate meta-monitoring which is a best practice (: 

I think the main questions is to nail down exact API - what we want to people use it for and if a nice metric exposed by the receiver to meta-monitoring cannot fulfil the same purpose. 

### Why not

* Metrics exposed by receiver could hold the same information?
* Is this data for admin / operator of Thanos or for Thanos user?

**Comment by yeya24** [comment]
@fpetkovski Any next step for this issue?

**Comment by fpetkovski** [comment]
Unfortunately I don't have capacity to work on this right now, I am primarily focused on the new engine. If no one has stepped up to tackle this, maybe we can close it for now?
